File: nodejs_backend/models/lecturer.model.js
const mongoose = require('mongoose');
const { Schema } = mongoose;
const db = require('../config/db');

// Định nghĩa schema và model cho sinh viên
const lecturerSchema = new Schema({
    lecturerID: {
        type: String,
        required: [true, "Student ID is required"],
        unique: true,
    },
    lecturerName: {
        type: String,
        required: [true, "Student Name is required"],
    },
    gender: {
        type: String,
        required: [true, "Gender is required"],
        enum: ['Nam', 'Nữ'],
    },
    birthDate: {
        type: Date,
        required: [true, "Birth Date is required"],
    },
}, { timestamps: true });

const Lecturer = db.model('lecturer', lecturerSchema);
module.exports = Lecturer;

File: flutter/lib/model/lecturerModel.dart


class Lecturer {
  final String id;
  final String lecturerID;
  final String lecturerName;
  final String gender;
  final DateTime birthDate;
  final String position;
  final String facultyID;

  Lecturer(
      {required this.id,
      required this.lecturerID,
      required this.lecturerName,
      required this.gender,
      required this.birthDate,
      required this.position,
      required this.facultyID});

  factory Lecturer.fromMap(Map<String, dynamic> json) {
    return Lecturer(
      id: json['_id'],
      lecturerID: json['lecturerID'],
      lecturerName: json['lecturerName'],
      gender: json['gender'],
      birthDate: DateTime.parse(json['birthDate']),
      position: json['position'],
      facultyID: json['facultyID'],
    );
  }

  @override
  String toString() {
    return '${id} - ${lecturerID} - ${lecturerName} - ${gender} - ${birthDate} - ${position} - ${facultyID}';
  }
}
